ReactAndroid/src/main/java/com/facebook/react/modules/appstate/AppStateModule.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+package com.facebook.react.modules.appstate;
+
+import com.facebook.react.bridge.Arguments;
+import com.facebook.react.bridge.Callback;
+import com.facebook.react.bridge.LifecycleEventListener;
+import com.facebook.react.bridge.Promise;
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.bridge.ReactContextBaseJavaModule;
+import com.facebook.react.bridge.ReactMethod;
+import com.facebook.react.bridge.WritableMap;
+
+import static com.facebook.react.modules.core.DeviceEventManagerModule.RCTDeviceEventEmitter;
+
+public class AppStateModule extends ReactContextBaseJavaModule
+        implements LifecycleEventListener {
+
+  private String mAppState = "active";
+
+  public AppStateModule(ReactApplicationContext reactContext) {
+    super(reactContext);
+  }
+
+  @Override
+  public String getName() {
+    return "AppState";
+  }
+
+  @Override
+  public void initialize() {
+    getReactApplicationContext().addLifecycleEventListener(this);
+  }
+
+  @ReactMethod
+  public void getCurrentAppState(Promise promise) {
+    promise.resolve(createAppStateEventMap());
+  }
+
+  @Override
+  public void onHostResume() {
+    mAppState = "active";
+    sendAppStateEvent();
+  }
+
+  @Override
+  public void onHostPause() {
+    mAppState = "background";
+    sendAppStateEvent();
+  }
+
+  @Override
+  public void onHostDestroy() {
+
```

Comment:
Please add the comment from https://gist.github.com/mkonicek/039f35748c0a073dd1b5